pallets/dex-api/src/lib.rs

```diff
@@ -331,21 +331,19 @@ impl<T: Config>
         T::DexInfoProvider::ensure_dex_exists(&filter.dex_id)?;
         Ok(supported_types
             .iter()
-            .filter_map(|source_type| {
-                if filter.matches_index(*source_type)
-                    && Self::can_exchange(
-                        &LiquiditySourceId::new(filter.dex_id, *source_type),
-                        input_asset_id,
-                        output_asset_id,
-                    )
+            .filter_map(|source_type| match filter.matches_index(*source_type) {
+                true if Self::can_exchange(
+                    &LiquiditySourceId::new(filter.dex_id, *source_type),
+                    input_asset_id,
+                    output_asset_id,
+                ) =>
                 {
                     Some(LiquiditySourceId::new(
                         filter.dex_id.clone(),
                         source_type.clone(),
                     ))
-                } else {
-                    None
                 }
+                _ => None,
```

Comment:
For what? `if-else` looks better for bool than `match`.
Looks redundant, please revert

Reply:
It's just that I'm a fan of such expressions. Reverted.